### PR TITLE
chore(web): support draw clipping for 3dtiles

### PR DESCRIPTION
**Pull request by airslice**
# Overview

This PR supports a new type of clipping on 3dtiles. It will generate side planes with a polygon and top/bottom plane with relative height (base height is the sampled terrain height of center point of the polygon).

## What I've done

- Add draw clipping to 3d tiles appearance `experimental_clipping.draw`
  - Basially i don't want to effect the original clipping box, all draw clipping needed is in this `draw` property, it will have higher priority campare with clipping box.
- Refactor some on sketch API to better fit the requirement of draw clipping.

NOTE: Concave polygon is not supported (from engine limit)

### CODE REVIEW

File: web/src/beta/lib/core/engines/Cesium/Feature/Tileset/DrawClippingPolygon/index.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    () =>
+      coordiantes
+        ? new PolygonHierarchy(coordiantes.map(c => Cartesian3.fromDegrees(c[0], c[1], c[2] ?? 0)))
+        : undefined,
+    [coordiantes ?? []],
+    isEqual,
+  );
+
+  const strokeColor = useMemo(() => toColor("#00bebe"), []);
+  const fillColor = useMemo(() => toColor("#00bebe11"), []);
```
Comment (keiya01) [inline web/src/beta/lib/core/engines/Cesium/Feature/Tileset/DrawClippingPolygon/index.tsx:35]: Should these color be customizable?

File: web/src/beta/lib/core/engines/Cesium/Sketch/ExtrudedPolygonEntity.tsx [hunk 2]
```diff
@@ -53,12 +55,18 @@ export const ExtrudedPolygonEntity: FC<ExtrudedPolygonEntityProps> = ({
         extrudedHeight,
         extrudedHeightReference: HeightReference.RELATIVE_TO_GROUND,
         fill: true,
+        outline: true,
+        outlineWidth: 1,
```
Comment (keiya01) [inline web/src/beta/lib/core/engines/Cesium/Sketch/ExtrudedPolygonEntity.tsx:59]: Should the outline be customizable from plugin API?

**Review by keiya01 — APPROVED** [review]

File: web/src/beta/lib/core/engines/Cesium/Feature/Tileset/DrawClippingPolygon/index.tsx (hunk 1 above)
Comment (airslice) [reply web/src/beta/lib/core/engines/Cesium/Feature/Tileset/DrawClippingPolygon/index.tsx]: Yes better have that. I updated the style support in appearance.

File: web/src/beta/lib/core/engines/Cesium/Sketch/ExtrudedPolygonEntity.tsx (hunk 2 above)
Comment (airslice) [reply web/src/beta/lib/core/engines/Cesium/Sketch/ExtrudedPolygonEntity.tsx:59]: For the sketch API i want to have a refactor in another PR. So i'll skip this one here.

**Pull request merged**